src/shared/libcrypt-util.c

```diff
@@ -90,6 +90,32 @@ int make_salt(char **ret) {
 #endif
 }
 
+#if !HAVE_CRYPT_RA
+/* Provide a poor man's fallback that uses a fixed size buffer. */
+
+static char* systemd_crypt_ra(const char *phrase, const char *setting, void **data, int *size) {
+        _cleanup_(erase_and_freep) void *cd = NULL;
+
```

Comment:
mayb add a comment why we bother with dynamic memory here, i.e. reference that comment in the man page